### PR TITLE
fix: add retry to s3 uploads

### LINKED ISSUE #32: Add exponential backoff to S3 uploads in upload.ts
S3.putObject() likes to return 503s intermittently (but often enough to make our CD need painful reruns). [retried](https://deno.land/x/retried@1.0.1) or another library wrapping that function should help a lot.

### PR DESCRIPTION
**Pull request by lino-levan**
Closes #32.

I removed the unnecessary dependency on `retried` since the deno standard library already has a method to do exactly that (I actually authored it 😛).

### CODE REVIEW

File: import-map.json [hunk 1]
```diff
@@ -9,12 +9,11 @@
     "types": "https://raw.githubusercontent.com/teaxyz/cli/v0.19.1/src/types.ts",
     "hooks/": "https://raw.githubusercontent.com/teaxyz/cli/v0.19.1/src/hooks/",
     "utils/": "https://raw.githubusercontent.com/teaxyz/cli/v0.19.1/src/utils/",
-    "deno/": "https://deno.land/std@0.156.0/",
+    "deno/": "https://deno.land/std@0.172.0/",
```
Comment (mxcl) [inline import-map.json:12]: cor. we behind

**Comment by mxcl** [comment]
Thank you! I'll have @jhheider review since it effects his areas more than mine.

**Review by mxcl — APPROVED** [review]

**Comment by jhheider** [comment]
Very nice. Bumping the deno version breaks the sha256 hasher, it looks like, so we'll need to migrate that to whatever the new standard is. Thanks for writing the retry function!

**Comment by lino-levan** [comment]
Sure, let me fix the sha256 hasher.

**Comment by lino-levan** [comment]
It looks like we actually need to update the deno dependencies in tea/cli before this can land. A bit dubious that this and the cli are tied together in my opinion but /shrug.

I'll make a PR

**Comment by jhheider** [comment]
> It looks like we actually need to update the deno dependencies in tea/cli before this can land. A bit dubious that this and the cli are tied together in my opinion but /shrug.
> 
> I'll make a PR

Historically, it was tighter, since this was working with the exposed API bits in TypeScript. Eventual plan is to expose the public API via a `libtea`, so we don't have these kinds of versioning conflicts.

**Comment by mxcl** [comment]
> It looks like we actually need to update the deno dependencies in tea/cli before this can land. A bit dubious that this and the cli are tied together in my opinion but /shrug.

yeah as Jacob says, it's not desired to stay this way. Was a result of moving fast.

**Comment by mxcl** [comment]
One thing we should probably do is have the import-map specified here be the URL to tea/cli’s import-map. Provided we use a tag’d version it’s safe from revlock.

I noticed the VSCode extension doesn’t like this. And it means updating *every* script’s args list.

**Comment by lino-levan** [comment]
@mxcl, this should work now. Please approve the workflow.

> One thing we should probably do is have the import-map specified here be the URL to tea/cli’s import-map. Provided we use a tag’d version it’s safe from revlock.

~Doesn't this have dependencies outside of tea/cli?~ Nevermind, it doesn't. This makes sense.

> And it means updating every script’s args list.

?

**Comment by mxcl** [comment]
> > And it means updating every script’s args list.
> ?

All our scripts specify the import map, but probs a `deno.json` is all we really need.

**Comment by lino-levan** [comment]
That's embarrassing, I forgot what the input was to my own function

**Pull request merged**